libfwupdplugin/fu-kernel.c

```diff
@@ -218,3 +219,40 @@ fu_kernel_reset_firmware_search_path(GError **error)
 
 	return fu_kernel_set_firmware_search_path(contents, error);
 }
+
+/**
+ * fu_kernel_get_proc_cmdline:
+ * @error: (nullable): optional return location for an error
+ *
+ * Loads all the kernel /proc/cmdline key/values into a hash table.
+ *
+ * Returns: (transfer container) (element-type utf8 utf8): keys from the kernel command line
+ *
+ * Since: 1.8.5
+ **/
+GHashTable *
+fu_kernel_get_proc_cmdline(GError **error)
```

Comment:
I think this should have some `#ifndef _WIN32` guard

Reply:
Does macos have /proc/cmdline? The one in fu-engine.c has #ifdef __linux__

Reply:
I'm pretty sure `/proc/cmdline` is a linux only concept.
Windows has https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getcommandlinea, but that's not really the same thing as `/proc/cmdline`, it's more like `/proc/self/cmdline`.

Reply:
Gotcha, thanks. Fixed.